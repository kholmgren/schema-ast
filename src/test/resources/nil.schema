definition resource {    
    permission empty = nilRefExpr
    permission another = foo + nilRefExpr + bar
    permission third = (a + b + nilRefExpr) - c - nilRefExpr
}