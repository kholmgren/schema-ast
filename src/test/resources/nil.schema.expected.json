{
  "@type" : "schema",
  "definitions" : [ {
    "@type" : "definition",
    "name" : "resource",
    "permissions" : [ {
      "@type" : "permission",
      "name" : "empty",
      "expression" : {
        "@type" : "nil"
      }
    }, {
      "@type" : "permission",
      "name" : "another",
      "expression" : {
        "@type" : "union",
        "left" : {
          "@type" : "union",
          "left" : {
            "@type" : "ref",
            "ref" : "foo"
          },
          "right" : {
            "@type" : "nil"
          }
        },
        "right" : {
          "@type" : "ref",
          "ref" : "bar"
        }
      }
    }, {
      "@type" : "permission",
      "name" : "third",
      "expression" : {
        "@type" : "exclude",
        "left" : {
          "@type" : "exclude",
          "left" : {
            "@type" : "union",
            "left" : {
              "@type" : "union",
              "left" : {
                "@type" : "ref",
                "ref" : "a"
              },
              "right" : {
                "@type" : "ref",
                "ref" : "b"
              }
            },
            "right" : {
              "@type" : "nil"
            }
          },
          "right" : {
            "@type" : "ref",
            "ref" : "c"
          }
        },
        "right" : {
          "@type" : "nil"
        }
      }
    } ]
  } ]
}